plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -337,11 +338,12 @@ func (s *SQLServer) Stop() {
 	}
 }
 
-func (s *SQLServer) gatherServer(pool *sql.DB, query Query, acc telegraf.Accumulator) error {
+func (s *SQLServer) gatherServer(pool *sql.DB, query Query, acc telegraf.Accumulator, connectionString string) error {
 	// execute query
 	rows, err := pool.Query(query.Script)
 	if err != nil {
-		return fmt.Errorf("script %s failed: %w", query.ScriptName, err)
+		serverName, databaseName := getConnectionIdentifiers(connectionString)
+		return fmt.Errorf("script %s failed: %w. ServerName: %s and DatabaseName: %s", query.ScriptName, err, serverName, databaseName)
```

Comment:
I would suggest to place the server and database names before the error message:
```
return fmt.Errorf("script %s failed for server %s and database %s: %w", query.ScriptName, serverName, databaseName, err)
```

Reply:
Done!